docs/arcadia-users-group/20220919-aws-s3-cli/lesson.md

```diff
@@ -0,0 +1,301 @@
+
+```{bash}
+aws s3 sync . s3://aug-workshop-demo/fmc/
+aws s3 ls s3://aug-workshop-demo/fmc/
+```
+
+Now, everything is in S3. Let's try it again:
+```{bash}
+aws s3 sync . s3://aug-workshop-demo/fmc/
+```
+
+Nothing happens! This is because all the files as they are on S3. Let's make a change to the `copy.txt` file:
+
+```{bash}
+echo "new copy" >> copy.txt
+aws s3 sync . s3://aug-workshop-demo/fmc/
+aws s3 ls s3://aug-workshop-demo/fmc/
+```
+
+### Downloading data from S3
+
+Our examples so far have been about uploading data to S3, but we can easily switch the direction of operations to download data from S3. The main change is the argument order. To download data from S3, we need S3 to be our source and local file system to be the target.
+
+Let's copy and move some data from S3 as a simple example:
+```{bash}
+aws s3 cp s3://aug-workshop-demo/fmc/copy.txt new-copy.txt
+aws s3 mv s3://aug-workshop-demo/fmc/move.txt new-move.txt
+aws s3 ls s3://aug-workshop-demo/fmc/
+```
+
+### --dryrun
+
+In using all of these commands, if a command writes or modifies data, it's a good idea to visualize what changes will be made to your local system or the S3 storage system before making them. For this the `--dryrun` flag is your friend.
+
+When used, it displays the operations that would be performed using the specified command without actually executing them. So, if you're in the wrong directory or about to modify or delete a file irreversibly by mistake, it allows you to catch these issues ahead of time. Let's see it in action:
+
+```{bash}
+aws s3 rm --dryrun --recursive s3://aug-workshop-demo/fmc/
+```
+
+## [Optional] Advanced usage with s5cmd
+
+[s5cmd](https://github.com/peak/s5cmd) is an unofficial tool to interact with AWS S3 through the command-line. In this section, we'll talk about the installation instructions and the differences to the official AWS S3 CLI. But first, why do we care about `s5cmd`?
+
+### Why?
+
+Great news: if your machine is configured to work with the AWS CLI, it's by default configured to work with s5cmd! It has two benefits over the official S3 CLI:
+1. It's much faster. This is due to effective parallelization and bandwidth saturation. See [this blog post](https://joshua-robinson.medium.com/s5cmd-for-high-performance-object-storage-7071352cc09d) for benchmarking data.
+![](s5cmd-benchmark.png)
+2. It also is compatible with Google Cloud Storage (GCS) in case you have to work with any databases that are hosted on GCS (think: Alphafold).
+
+### Downloading and installing the s5cmd CLI
+
+```{bash}
+wget https://github.com/peak/s5cmd/releases/download/v2.0.0/s5cmd_2.0.0_Linux-64bit.tar.gz
+tar -xvf s5cmd_2.0.0_Linux-64bit.tar.gz
```

Comment:
Are these instructions specific to linux? What would the mac installation instructions be?